### PR TITLE
feat(ctb): `PreimageOracle` LPP Diff Testing

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds some differential testing for the large preimage proposal route in the `PreimageOracle`.

### CODE REVIEW

File: packages/contracts-bedrock/test/cannon/PreimageOracle.t.sol [hunk 1]
```diff
@@ -776,6 +776,60 @@ contract PreimageOracle_LargePreimageProposals_Test is Test {
...
+        // Add the leaves to the tree with corrupted state commitments.
+        LibKeccak.StateMatrix memory matrixA;
+        bytes32[] memory stateCommitments = _generateStateCommitments(matrixA, data);
+        for (uint256 i = _lastCorrectLeafIdx + 1; i < stateCommitments.length; i++) {
+            stateCommitments[i] = 0;
+        }
+        oracle.addLeavesLPP(TEST_UUID, data, stateCommitments, true);
+
+        // Construct the leaf preimage data for the blocks added and corrupt the state commitments.
+        data = new bytes(136 * _numBlocks);
```
Comment (clabby) [inline packages/contracts-bedrock/test/cannon/PreimageOracle.t.sol]: There is some spooky action at a distance happening here.

**Comment by refcell** [comment]
@dependabot rebase

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #9147** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9147?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9147?utm_source=stack-comment).

**Review by refcell — APPROVED** [review]
Good stuff, thanks for fixing this up 👑

**Comment by ajsutton** [comment]
Nice. :)

**Pull request merged**